### PR TITLE
Release sprint 7

### PR DESCRIPTION
**Pull request by KVISDAOWNER**
Work from sprint 6 and sprint 7

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,9 +2,11 @@ version: 2.1
 
 orbs:
  android: circleci/android@0.2.0
+ windows: circleci/windows@2.2.0
+
 
 jobs:
-  build:
+  build_APP:
     executor: android/android
```
Comment (Sworgon) [inline .circleci/config.yml:10]: What's the difference between build and build_APP

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/apis/DeadCrumbsApi.kt [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+     * 
+     * @param username The username of the user. 
+     * @return User
+     */
+    @Suppress("UNCHECKED_CAST")
+    fun getUser(username: kotlin.String): User {
+        
+        val localVariableConfig = RequestConfig(
+                RequestMethod.GET,
+                "/User/{username}".replace("{" + "username" + "}", "$username")
```
Comment (Sworgon) [inline DeadCrumbs/app/src/main/java/dead/crumbs/client/apis/DeadCrumbsApi.kt:57]: Why write "{" + "username" + "}", when you could just write "{username}"?

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/apis/DeadCrumbsApi.kt [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                RequestMethod.POST,
+                "/Location"
+        )
+        val response = request<Any?>(
+                localVariableConfig,
+                localVariableBody
+        )
+
+        return when (response.responseType) {
+            ResponseType.Success -> Unit
```
Comment (Sworgon) [inline DeadCrumbs/app/src/main/java/dead/crumbs/client/apis/DeadCrumbsApi.kt:142]: What is Unit?

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/apis/DeadCrumbsApi.kt [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+     * @param dist step length 
+     * @param timeStamp The time. 
+     * @return Location
+     */
+    @Suppress("UNCHECKED_CAST")
+    fun updateLocation(username: kotlin.String, orientation: kotlin.Double, dist: kotlin.Double, timeStamp: kotlin.String): Location {
+        
+        val localVariableConfig = RequestConfig(
+                RequestMethod.POST,
+                "/Location/{username}/{orientation}/{dist}/{timeStamp}".replace("{" + "username" + "}", "$username").replace("{" + "orientation" + "}", "$orientation").replace("{" + "dist" + "}", "$dist").replace("{" + "timeStamp" + "}", "$timeStamp")
```
Comment (Sworgon) [inline DeadCrumbs/app/src/main/java/dead/crumbs/client/apis/DeadCrumbsApi.kt:188]: Same as above

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt [hunk 5]
```diff
@@ -0,0 +1,42 @@
+@file:Suppress("unused")
+package io.swagger.client.infrastructure
+
+import java.lang.RuntimeException
+
+open class ClientException : RuntimeException {
+
+    /**
+     * Constructs an [ClientException] with no detail message.
```
Comment (Sworgon) [inline DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt:9]: a*

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+     */
+    constructor() : super()
+
+    /**
+     * Constructs an [ClientException] with the specified detail message.
```
Comment (Sworgon) [inline DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt:14]: same as above

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+    companion object {
+        private const val serialVersionUID: Long = 123L
+    }
+}
+
+open class ServerException : RuntimeException {
+
+    /**
+     * Constructs an [ServerException] with no detail message.
```
Comment (Sworgon) [inline DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt:28]: same as above

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+     */
+    constructor() : super()
+
+    /**
+     * Constructs an [ServerException] with the specified detail message.
```
Comment (Sworgon) [inline DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt:33]: same as above

File: DeadCrumbs/app/src/main/java/dead/crumbs/ui/MapsViewModel.kt [hunk 9]
```diff
@@ -1,24 +1,36 @@
...
-    lateinit var map: GoogleMap
-
-    var markerList = mutableListOf<Marker>()
+class MapsViewModel (private val locationRepository: LocationRepository,
+                     private val userRepository: UserRepository) : ViewModel(){
 
+    lateinit var map: GoogleMap
+    var markerList = mutableListOf<Marker>() // the list of markers that are displayed on the map
     var mapIsInitialized = false
+    val PERMISSION_ID = 42 //a value to check if the users gives permission to what we ask for
```
Comment (Sworgon) [inline DeadCrumbs/app/src/main/java/dead/crumbs/ui/MapsViewModel.kt:31]: Why is this needed?

File: DeadCrumbs/app/src/main/java/dead/crumbs/ui/MapsViewModel.kt [hunk 10]
```diff
@@ -27,83 +39,257 @@ class MapsViewModel (private val mapsRepository: MapsRepository) : ViewModel(){
...
+            PERMISSION_ID
+        )
     }
 
-    //Updates the location of a marker based on the length of
-    //the newest detected step and the current orientation of the marker.
-    //Expects length to be in meters.
-    private fun moveMarker(marker: Marker, mDist: Double){
-        val heading = Math.toRadians(marker.rotation.toDouble())
-        val R = 6378.1 //Radius of the Earth
```
Comment (Sworgon) [inline DeadCrumbs/app/src/main/java/dead/crumbs/ui/MapsViewModel.kt:89]: The Earth is flat and therefore has no radius

File: aspnet-server-1/IO.Swagger.sln [hunk 11]
```diff
@@ -1,25 +1,25 @@
...
 	EndGlobalSection
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
-		{A0ED41D7-B824-4121-83C0-520E9867900B}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
-		{A0ED41D7-B824-4121-83C0-520E9867900B}.Debug|Any CPU.Build.0 = Debug|Any CPU
-		{A0ED41D7-B824-4121-83C0-520E9867900B}.Release|Any CPU.ActiveCfg = Release|Any CPU
-		{A0ED41D7-B824-4121-83C0-520E9867900B}.Release|Any CPU.Build.0 = Release|Any CPU
+		{034A8537-EF72-4510-AD96-D13914F00832}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{034A8537-EF72-4510-AD96-D13914F00832}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{034A8537-EF72-4510-AD96-D13914F00832}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{034A8537-EF72-4510-AD96-D13914F00832}.Release|Any CPU.Build.0 = Release|Any CPU
```
Comment (Sworgon) [inline aspnet-server-1/IO.Swagger.sln:17]: Good choice

File: aspnet-server-1/client-api-spec.yaml [hunk 12]
```diff
@@ -0,0 +1,237 @@
...
+
+  /Location/{username}:
+    get:
+      summary: returns the location associated with username
+      operationId: getLocation
+      parameters:
+        - $ref: '#/components/parameters/username'
+      responses:
+        200:
+          description: Succesfull request! Returns a location represented as JSON
```
Comment (Sworgon) [inline aspnet-server-1/client-api-spec.yaml:61]: I thought we went through this. Successful*

File: aspnet-server-1/src/IO.Swagger/Controllers/DeadCrumbsApi.cs [hunk 13]
```diff
@@ -0,0 +1,369 @@
...
+            if (user == null)
+            {
+                return StatusCode(404, $"No user found called \"{username}\"!");
+            }
+            var result = new ObjectResult(user);
+            return result;
+        }
+
+        /// <summary>
+        /// returns all user as a list
```
Comment (Sworgon) [inline aspnet-server-1/src/IO.Swagger/Controllers/DeadCrumbsApi.cs:94]: users*

**Review by User12837985123 — APPROVED** [review]
Everything looks and works fantastic!!!! Especially you Kris <3
I did not test Bluetooth sync, as it does not work on my phone :/

**Review by Sworgon — APPROVED** [review]
Generally looks good, but there are a few mistakes

File: .circleci/config.yml (hunk 1 above)
Comment (KVISDAOWNER) [reply .circleci/config.yml:10]: the difference is _APP

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/apis/DeadCrumbsApi.kt (hunk 2 above)
Comment (KVISDAOWNER) [reply DeadCrumbs/app/src/main/java/dead/crumbs/client/apis/DeadCrumbsApi.kt:57]: This is autogenerated

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/apis/DeadCrumbsApi.kt (hunk 3 above)
Comment (KVISDAOWNER) [reply DeadCrumbs/app/src/main/java/dead/crumbs/client/apis/DeadCrumbsApi.kt:142]: Unit = void

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/apis/DeadCrumbsApi.kt (hunk 4 above)
Comment (KVISDAOWNER) [reply DeadCrumbs/app/src/main/java/dead/crumbs/client/apis/DeadCrumbsApi.kt:188]: It is autogenerated

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt (hunk 5 above)
Comment (KVISDAOWNER) [reply DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt:9]: Auto generated

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt (hunk 6 above)
Comment (KVISDAOWNER) [reply DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt:14]: Auto generated

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt (hunk 8 above)
Comment (KVISDAOWNER) [reply DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt:33]: Auto generated

File: DeadCrumbs/app/src/main/java/dead/crumbs/ui/MapsViewModel.kt (hunk 9 above)
Comment (KVISDAOWNER) [reply DeadCrumbs/app/src/main/java/dead/crumbs/ui/MapsViewModel.kt:31]: No idea

File: DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt (hunk 7 above)
Comment (KVISDAOWNER) [reply DeadCrumbs/app/src/main/java/dead/crumbs/client/infrastructure/Errors.kt:28]: Autogenerated
